src/main/java/com/cdancy/jenkins/rest/parsers/CrumbParser.java

```diff
@@ -38,13 +39,24 @@ public Crumb apply(final HttpResponse input) {
         final int statusCode = input.getStatusCode();
         if (statusCode >= 200 && statusCode < 400) {
             try {
-                final String response = Strings2.toStringAndClose(input.getPayload().openStream());
-                return Crumb.create(response.split(":")[1], null);
+                return Crumb.create(crumbValue(input), sessionIdCookie(input));
             } catch (final IOException e) {
                 throw new RuntimeException(input.getStatusLine(), e);
             }
         } else {
             throw new RuntimeException(input.getStatusLine());
         }
     }
+
+    private static String crumbValue(HttpResponse input) throws IOException {
+        return Strings2.toStringAndClose(input.getPayload().openStream())
+                .split(":")[1];
+    }
+
+    private static String sessionIdCookie(HttpResponse input) {
+        return input.getHeaders().get(HttpHeaders.SET_COOKIE).stream()
+            .filter(c -> c.startsWith("JSESSIONID"))
```

Comment:
Can we turn JSESSIONID into a `static final String` constant and put it somewhere that makes the most sense? Believe there is a static-variables class around here somewhere

Reply:
I found the [com.cdancy.jenkins.rest.JenkinsConstants](https://github.com/cdancy/jenkins-rest/blob/v0.0.20/src/main/java/com/cdancy/jenkins/rest/JenkinsConstants.java#L23) class. Is that the one you had in mind?

Reply:
Yeah exactly

Reply:
done, see d0a16f2